optimum/exporters/onnx/__main__.py

```diff
@@ -146,14 +146,8 @@ def main_export(
         )
 
     original_task = task
-    # Infer the task
-    if task == "auto":
-        try:
-            task = TasksManager.infer_task_from_model(model_name_or_path)
-        except KeyError as e:
-            raise KeyError(
-                f"The task could not be automatically inferred. Please provide the argument --task with the task from {', '.join(TasksManager.get_all_tasks())}. Detailed error: {e}"
-            )
+
+    task = TasksManager.map_from_legacy(task)
```

Comment:
Insteqd of having this, I think we should define a `"task_synonyms"` dictionary in the `TasksManager`, allowing to specify either `sequence-classification` or `text-classification` for instance, not make it legacy but rather synonyms.